scanpy/external/tl/_gsea.py

```diff
@@ -0,0 +1,226 @@
+from typing import Optional, Tuple
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import scanpy.external as sce
+
+def gsea_barplot(
+    df: pd.DataFrame, 
+    score: Optional['str'] = 'nes',
+    score_cutoff: Optional[float] = None,
+    top_term: Optional[float] = 10,
+    figsize: Optional[Tuple[float, float]] = (6.5,6),
+    title: Optional['str'] = None,
+    x_label: Optional['str'] = None,
+    y_label: Optional['str'] = None,
+    out_dir: Optional['str'] = 'gseabarplot.png',
+    color = '#1f77b4', # matplotlib default: https://matplotlib.org/stable/users/dflt_style_changes.html#id1
+):
+  
+    """\
+    Horizontal barplot of GSEA results of genes by enrichment terms given a dataframe
+
+    Allows visualizations of either pval, nes, and fdr from outputs using either gseapy or fgsea.
+    Dataframe likely generated from scanpy.external.tl.gsea() function.
```

Comment:
sorry just because I don't understand it personally, what is the like "likely generated" part - is that the function that's being built with R integration?

Reply:
oops that was leftover from when I wrote the gseabarplot function before the actual gsea function, have removed it. the gsea_barplot() function _does_ take in the output from the gsea() function